From 4ee14a152413e569456d8648241bc4a7d46dc390 Mon Sep 17 00:00:00 2001
From: yangwei1 <yangwei1@eswincomputing.com>
Date: Thu, 12 Sep 2024 09:18:33 +0800
Subject: [PATCH 264/652] WIN2030-16287:feat:suppor tmp102 to get board
 temperature

Changelogs:
you can get baord temperature by sysfs name temp1_input in hwmon

Change-Id: I6a2185a701c7b71a06ec0b419a057aa6601da429
Signed-off-by: yangwei1 <yangwei1@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eic7702-evb-a1-d0.dts |  6 ++++++
 arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts    | 12 ++++++++++++
 arch/riscv/configs/eic7702_defconfig            |  1 +
 3 files changed, 19 insertions(+)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-evb-a1-d0.dts b/arch/riscv/boot/dts/eswin/eic7702-evb-a1-d0.dts
index 22a8b5764493..3bf4332d267a 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-evb-a1-d0.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-evb-a1-d0.dts
@@ -716,6 +716,12 @@ ina226@44 {
 		reg = <0x44>;
 		shunt-resistor = <1000>;
 	};
+	tmp102@48 {
+		compatible = "ti,tmp102";
+		reg = <0x48>;
+		label = "d0_board_tmp";
+		#thermal-sensor-cells = <1>;
+	};
 };
 
 &d0_aon_i2c1 {
diff --git a/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts b/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
index 6d1db9c6a713..d1c9d8cbd5b3 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-evb-a1.dts
@@ -1239,6 +1239,12 @@ ina226@44 {
 		reg = <0x44>;
 		shunt-resistor = <1000>;
 	};
+	tmp102@48 {
+		compatible = "ti,tmp102";
+		reg = <0x48>;
+		label = "d0_board_tmp";
+		#thermal-sensor-cells = <1>;
+	};
 };
 
 
@@ -1267,6 +1273,12 @@ ina226@44 {
 			reg = <0x44>;
 			shunt-resistor = <1000>;
 		};
+	tmp102@48 {
+		compatible = "ti,tmp102";
+		reg = <0x48>;
+		label = "d1_board_tmp";
+		#thermal-sensor-cells = <1>;
+	};
 };
 &d1_aon_i2c1 {
 	status = "disabled";
diff --git a/arch/riscv/configs/eic7702_defconfig b/arch/riscv/configs/eic7702_defconfig
index f0de31f666a0..1e9388335206 100644
--- a/arch/riscv/configs/eic7702_defconfig
+++ b/arch/riscv/configs/eic7702_defconfig
@@ -460,6 +460,7 @@ CONFIG_GPIO_DWAPB=y
 CONFIG_SENSORS_ESWIN_FAN_CONTROL=y
 CONFIG_SENSORS_ESWIN_PVT=y
 CONFIG_SENSORS_INA2XX=y
+CONFIG_SENSORS_TMP102=y
 CONFIG_SENSORS_PAC1934=y
 CONFIG_WATCHDOG=y
 CONFIG_DW_WATCHDOG=y
-- 
2.54.0

